### ISSUE TITLE
Support for add_scalars in log

### ISSUE THREAD

**Issue by GuHongyang** [williamFalcon/pytorch-lightning#361]
**Is your feature request related to a problem? Please describe.**
Sometimes we want to use add_scalars instead of add_scalar in the log results. For example, our loss function is a multitasking loss function, and we want to be able to display a graph of each loss function in the same graph.

**Comment by GuHongyang** [comment]
def training_step(self, batch, batch_nb):
        # REQUIRED
        x, y = batch
        y_hat = self.forward(x)
        loss = F.cross_entropy(y_hat, y)
        # tensorboard_logs = {'train_loss': loss}
        #We want to draw multiple loss function curves in the same coordinate system
        tensorboard_logs = {'train_loss1': loss,
                            'train_loss2': loss*2,
                            'train_loss3': loss*3,}
        return {'loss': loss, 'log': tensorboard_logs}
But in this way, we will get:
<img width="533" alt="屏幕快照 2019-10-12 下午4 34 04" src="https://user-images.githubusercontent.com/20881636/66698223-9c5a4f00-ed0e-11e9-995f-127897f13756.png">

**Comment by GuHongyang** [comment]
The root of the problem is that the log in test-tube only supports add_scalar and does not support add_scalars. In order to achieve the desired effect, I intend to change it like this:

    def training_step(self, batch, batch_nb):
        # REQUIRED
        x, y = batch
        y_hat = self.forward(x)
        loss = F.cross_entropy(y_hat, y)
        # tensorboard_logs = {'train_loss': loss}
        #We want to draw multiple loss function curves
        # tensorboard_logs = {'train_loss1': loss,
        #                     'train_loss2': loss*2,
        #                     'train_loss3': loss*3,}
        #modify like this
        tensorboard_logs = {'losses':{'train_loss1': loss,
                            'train_loss2': loss*2,
                            'train_loss3': loss*3,}}
        return {'loss': loss, 'log': tensorboard_logs}

Not only that, but you also need to change the “log” function in test-tube/log.py to:

<img width="821" alt="屏幕快照 2019-10-12 下午4 43 55" src="https://user-images.githubusercontent.com/20881636/66698340-d0823f80-ed0f-11e9-9afa-0503b5c2a1a7.png">

Then we can get
<img width="410" alt="屏幕快照 2019-10-12 下午4 48 47" src="https://user-images.githubusercontent.com/20881636/66698379-366ec700-ed10-11e9-888a-43dba1f4a101.png">

**Comment by GuHongyang** [comment]
Is this scheme feasible?

**Comment by williamFalcon** [comment]
great suggestion! would welcome this PR!

**Comment by GuHongyang** [comment]
@williamFalcon 
Hello, it is better to use the add_image(s) to visualize some intermediate results during the training process! For example, when using the visual attention mechanism, we hope to see the heat map of attention during the training process.

**Comment by GuHongyang** [comment]
A desirable solution for using the above features is to add a new metric such as "log_imgs_metrics".
<img width="600" alt="屏幕快照 2019-10-12 下午10 03 51" src="https://user-images.githubusercontent.com/20881636/66702638-646a0080-ed3c-11e9-973c-55d3ed7034a2.png">
Do you think this kind of function is necessary?

**Comment by williamFalcon** [comment]
you can add anything you want during training... look at the GAN example. 

But generally you can just do:
self.logger.experiment.add_image
etc....

access self.logger.experiment from anywhere in your lightning module

**Comment by GuHongyang** [comment]
In gan.py it is trained with a single GPU. However, in multi-GPU (dp), add_image only displays (batch_size/nGPU) images, but not the entire batch_size image.

**Comment by williamFalcon** [comment]
correct. that’s by design. you wouldn’t see a difference otherwise because you are sampling from the model which on every gpu has the same weights, so it doesn’t matter
